contractsV5/bridges/bancor/KyberBancorReserve.sol

```diff
@@ -0,0 +1,232 @@
+        require(_bancorEth != address(0), "constructor: bancorEth address is missing");
+        require(_bancorToken != address(0), "constructor: bancorToken address is missing");
+        require(_admin != address(0), "constructor: admin address is missing");
+        require(_feeBps < BPS, "constructor: fee is too big");
+
+        bancorNetwork = IBancorNetwork(_bancorNetwork);
+        bancorToken = IERC20(_bancorToken);
+        bancorEth = IERC20(_bancorEth);
+
+        kyberNetwork = _kyberNetwork;
+        feeBps = _feeBps;
+        admin = _admin;
+        tradeEnabled = true;
+
+        require(bancorToken.approve(address(bancorNetwork), 2 ** 255));
+    }
+
+    function() external payable { }
+
+    function getConversionRate(IERC20 src, IERC20 dest, uint srcQty, uint) public view returns(uint) {
+        if (!tradeEnabled) return 0;
+        if (srcQty == 0) { return 0; }
+
+        if (src != ETH_TOKEN_ADDRESS && dest != ETH_TOKEN_ADDRESS) {
+            return 0; // either src or dest must be ETH
+        }
+        IERC20 token = src == ETH_TOKEN_ADDRESS ? dest : src;
+        if (token != bancorToken) { return 0; } // not BNT token
+
+        IERC20[] memory path = getConversionPath(src, dest);
+
+        uint destQty;
+        (destQty, ) = bancorNetwork.getReturnByPath(path, srcQty);
+
+        // src and dest can be only BNT or ETH
+        uint rate = calcRateFromQty(srcQty, destQty, ETH_BNT_DECIMALS, ETH_BNT_DECIMALS);
+
+        rate = valueAfterReducingFee(rate);
+
+        return rate;
+    }
+
+    event TradeExecute(
+        address indexed sender,
+        address src,
+        uint srcAmount,
+        address destToken,
+        uint destAmount,
+        address payable destAddress
+    );
+
+    function trade(
+        IERC20 srcToken,
+        uint srcAmount,
+        IERC20 destToken,
+        address payable destAddress,
+        uint conversionRate,
+        bool validate
+    )
+        public
+        payable
+        returns(bool)
+    {
+
+        require(tradeEnabled, "trade: trade is not enabled");
+        require(msg.sender == kyberNetwork, "trade: sender is not network");
+        require(srcToken == ETH_TOKEN_ADDRESS || destToken == ETH_TOKEN_ADDRESS, "trade: src or dest must be ETH");
+        require(srcToken == bancorToken || destToken == bancorToken, "trade: src or dest must be BNT");
+
+        require(doTrade(srcToken, srcAmount, destToken, destAddress, conversionRate, validate), "trade: doTrade function returns false");
+
+        return true;
+    }
+
+    event KyberNetworkSet(address kyberNetwork);
+
+    function setKyberNetwork(address _kyberNetwork) public onlyAdmin {
+        require(_kyberNetwork != address(0), "setKyberNetwork: kyberNetwork address is missing");
+
+        kyberNetwork = _kyberNetwork;
+        emit KyberNetworkSet(_kyberNetwork);
+    }
+
+    event BancorNetworkSet(address _bancorNetwork);
```

Comment:
inconsistent spacing.
please add space here after event deceleration
or remove in other places

Reply:
Fixed